### PR TITLE
setup-java v3

### PR DESCRIPTION
**Pull request by sullis**
---------
### Progress
- [x] Change must not contain extraneous whitespace
- [ ] Change must be properly reviewed (1 review required, with at least 1 [Committer](https://openjdk.org/bylaws#committer))

### Errors
&nbsp;⚠️ OCA signatory status must be verified
&nbsp;⚠️ The pull request body must not be empty.

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jmh pull/73/head:pull/73` \
`$ git checkout pull/73`

Update a local copy of the PR: \
`$ git checkout pull/73` \
`$ git pull https://git.openjdk.org/jmh pull/73/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 73`

View PR using the GUI difftool: \
`$ git pr show -t 73`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jmh/pull/73.diff">https://git.openjdk.org/jmh/pull/73.diff</a>

</details>

### CODE REVIEW

**Comment by shipilev** [comment]
Yeah, need OCA on file. 

What is the benefit of switching to v3 at this time?

File: .github/workflows/pre-integration.yml [hunk 1]
```diff
@@ -21,11 +21,12 @@ jobs:
...
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
     - name: Set up JDK ${{ matrix.java }}
-      uses: actions/setup-java@v1
+      uses: actions/setup-java@v3
       with:
         java-version: ${{ matrix.java }}
+        distribution: 'temurin'
     - name: Cache Maven packages
       uses: actions/cache@v2
```
Comment (jvirtanen) [inline .github/workflows/pre-integration.yml:31]: Now you could replace `actions/cache@v2` usage entirely by [specifying the input `cache: 'maven'` to `actions/setup-java@v3`](https://github.com/actions/setup-java#caching-maven-dependencies).

File: .github/workflows/pre-integration.yml [hunk 2]
```diff
@@ -21,11 +21,12 @@ jobs:
     name: Test JDK ${{ matrix.java }}, ${{ matrix.os }}, ${{ matrix.profile }}
 
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
     - name: Set up JDK ${{ matrix.java }}
-      uses: actions/setup-java@v1
+      uses: actions/setup-java@v3
```
Comment (TWiStErRob) [inline .github/workflows/pre-integration.yml:26]: > What is the benefit of switching to v3 at this time? -- @shipilev 

It runs on newer version of infrastructure (e.g. Node 12 -> 16), which might have performance improvements, vulnerability fixes etc.

I would ask: what's the reason to keep using an unmaintained almost 1 year old version? (really curious)

Comment (shipilev) [reply .github/workflows/pre-integration.yml:26]: Well, mostly "if it is not broken, don't fix it", but yes, we should switch to actively maintained version.

**Comment by shipilev** [comment]
This is subsumed by #84.

File: .github/workflows/pre-integration.yml (hunk 2 above)
Comment (TWiStErRob) [reply .github/workflows/pre-integration.yml:26]: @shipilev funnily enough: it's "broken" now ...

See warnings on https://github.com/shipilev/jmh/actions/runs/3243894873

https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/

Comment (shipilev) [reply .github/workflows/pre-integration.yml:26]: I'll handle it in #84.

**Comment by shipilev** [comment]
Consider closing this PR and test #84? Thanks.

File: .github/workflows/pre-integration.yml (hunk 2 above)
Comment (TWiStErRob) [reply .github/workflows/pre-integration.yml:26]: Yep 👍🏻, just wanted to follow the conversation here :)

**Pull request closed without merging**